### PR TITLE
add centroid implementation for MultiPoint

### PR DESCRIPTION
**Pull request by nberard**
Adds missing implementation of centroid on MultiPoint struct.

### CODE REVIEW

File: geo/src/algorithm/centroid.rs [hunk 1]
```diff
@@ -244,6 +244,40 @@ where
...
+{
+    type Output = Option<Point<T>>;
+
+    fn centroid(&self) -> Self::Output {
+        if self.0.is_empty() {
+            return None;
+        }
+        let sum = self.0.iter().fold(
+            Point::new(T::zero(), T::zero()),
+            |a: Point<T>, b: &Point<T>| Point::new(a.x() + b.x(), a.y() + b.y()),
```
Comment (urschrei) [inline geo/src/algorithm/centroid.rs]: do we need `b` to be **`&`**`Point<T>`?

Comment (TeXitoi) [reply geo/src/algorithm/centroid.rs]: `.iter()` returns a `Iterator<Item=&Point<T>>` thus you can't really do any other thing (type inference might work here).

**Comment by nberard** [comment]
the test is in the comment ;)

**Comment by TeXitoi** [comment]
That's OK for me, @urschrei I let you merge if everything is good on your side.

**Comment by frewsxcv** [comment]
https://github.com/Stoeoef/rstar/commit/d4ca14e29a9d4b567e1f80a9a9296bd8b0a7adc7

**Pull request merged**